test/testchecksumvalidator.cpp

```diff
@@ -84,10 +86,11 @@ using namespace OCC::Utility;
         QFileInfo fi(file);
         QVERIFY(fi.exists());
 
-        QFile fileDevice(file);
-        fileDevice.open(QIODevice::ReadOnly);
-        QByteArray sum = calcMd5(&fileDevice);
-        fileDevice.close();
+        QSharedPointer<QFile> fileDevice(new QFile(file));
+        const auto sharedFile = QSharedPointer<QIODevice>(fileDevice);
```

Comment:
to understand why this is not going to cause a double delete, I had to run it under valgrind (memory checking tool) and read the Qt doc
maybe it would be much simpler to do
```
        QSharedPointer<QIODevice> sharedFile(new QFile(file));
```